### PR TITLE
[Tabs] Add accessibility traits support

### PR DESCRIPTION
**Pull request by mikefan1991**
Add UIAccessibility traits to item view.

**Testing**
accessibilityTraits
Steps to reproduce

1. Update the example to have a custom accessibilityTraits for one of the tabBarItems
    1. Optionally have it change those traits in the `tabBarView:didSelectItem:` delegate method.
2. item1.accessibilityTraits = @"Hello world";
3. Launch on a device
4. Select the item with Voiceover on
5. Navigate to the other items with titles but no custom value.

### CODE REVIEW

File: components/Tabs/src/TabBarView/MDCTabBarView.m [hunk 1]
```diff
@@ -103,6 +104,7 @@ - (void)setItems:(NSArray<UITabBarItem *> *)items {
     itemView.accessibilityLabel = item.accessibilityLabel;
     itemView.accessibilityHint = item.accessibilityHint;
     itemView.accessibilityIdentifier = item.accessibilityIdentifier;
+    itemView.accessibilityTraits = item.accessibilityTraits;
```
Comment (romoore) [inline components/Tabs/src/TabBarView/MDCTabBarView.m]: I think this should default to `.button`:

```objc
itemView.accessibilityTraits = (item.accessibilityTraits == UIAccessibilityTraitNone) : UIAccessibilityTraitButton : item.accessibilityTraits;
```

**Review by romoore — CHANGES_REQUESTED** [review]
Looks like it needs to be updated to toggle `UIAccessibilityTraitSelected` trait during selection/unselection.  Should also be sure to default to `.button` either in the view or during assignment of the initial traits.

File: components/Tabs/src/TabBarView/MDCTabBarView.m (hunk 1 above)
Comment (mikefan1991) [reply components/Tabs/src/TabBarView/MDCTabBarView.m]: Done.

**Review by leonmz — COMMENTED** [review]
I thought we can add a unit tests to verify the behavior of the accessibility traits.

**Comment by mikefan1991** [comment]
The unit test is not feasible because the item views are private in the tab bar view.

**Comment by romoore** [comment]
The private nature is not the problem.  UIAccessibility does not function correctly outside of an application + view hierarchy. The PR should list manual testing steps that are taken (and can be reproduced) to verify the behavior. 

https://github.com/material-components/material-components-ios/pull/7729 is a good example of how to write manual testing for these attributes.

**Review by romoore — APPROVED** [review]

**Pull request merged**